lib/Arch/Arch.cpp

```diff
@@ -27,6 +29,54 @@ bool RegisterConstraint::ContainsVariant(const std::string& name) const {
   return false;
 }
 
+// Factory method that returns the correct calling convention for a given
+// architecture
+std::unique_ptr<CallingConvention> CallingConvention::CreateCCFromArch(
+    const remill::Arch *arch) {
+  switch (arch->arch_name) {
+    case remill::kArchInvalid: {
+      LOG(FATAL) << "Invalid architecture: "
+                 << remill::GetArchName(arch->arch_name);
+      break;
+    }
+    case remill::kArchX86: {
+      if (arch->os_name == remill::kOSmacOS ||
+          arch->os_name == remill::kOSLinux) {
+        return std::make_unique<X86_C>(arch);
+      } else {
+        LOG(FATAL) << "Unsupported (arch, os) pair: "
+                   << remill::GetArchName(arch->arch_name) << " "
+                   << remill::GetOSName(arch->os_name);
+      }
+      break;
+    }
+    case remill::kArchAMD64: {
+      if (arch->os_name == remill::kOSmacOS ||
+          arch->os_name == remill::kOSLinux) {
+        return std::make_unique<X86_64_SysV>(arch);
+      } else {
+        LOG(FATAL) << "Unsupported (arch, os) pair: "
+                   << remill::GetArchName(arch->arch_name) << " "
+                   << remill::GetOSName(arch->os_name);
+      }
+      break;
+    }
+
+    // Fallthrough for unsupported architectures
+    case remill::kArchX86_AVX:
```

Comment:
I think you should be able to handle the `_AVX` and `_AVX512` variants in the same way? Really, they will tell you whether or not you use `xmm*`, `ymm*`, or `zmm*`.

Reply:
Fixed in 455a2fc